synapse/handlers/room.py

```diff
@@ -586,7 +586,10 @@ async def clone_existing_room(
             user_id,
             {
                 "creation_content": creation_content,
-                "initial_state": list(initial_state.items()),
+                "initial_state": [
+                    {"type": state[0][0], "state_key": state[0][1], "content": state[1]}
+                    for state in initial_state.items()
+                ],
```

Comment:
If we add some types to where we define `initial_state: MutableStateMap = {}`

Then we can do this:

```suggestion
                "initial_state": [
                    {
                        "type": state_key[0],
                        "state_key": state_key[1],
                        "content": event_content,
                    }
                    for state_key, event_content in initial_state.items()
                ],
```